### PR TITLE
[4.0] Update null date check in blog layout

### PR DESCRIPTION
**Pull request by SharkyKZ**
Pull Request for Issue #26515

### Summary of Changes

Updates layout to check for real null date.

### Testing Instructions

Create some articles without publish down date.
Open category blog view.

### Expected result

No `Expired` badge under article title.

### Actual result

`Expired` badge shown.

### Documentation Changes Required

No.

### CODE REVIEW

File: layouts/joomla/content/blog_style_default_item_title.php [hunk 1]
```diff
@@ -42,9 +42,7 @@
 			<span class="badge badge-warning"><?php echo Text::_('JNOTPUBLISHEDYET'); ?></span>
 		<?php endif; ?>
 
-		<?php if ($displayData->publish_down != Factory::getDbo()->getNullDate()
-			&& (strtotime($displayData->publish_down) < strtotime(Factory::getDate()))
-		) : ?>
+		<?php if ($displayData->publish_down !== null && strtotime($displayData->publish_down) < strtotime(Factory::getDate())) : ?>
```
Comment (infograf768) [inline layouts/joomla/content/blog_style_default_item_title.php:45]: Normalize code?
In other files it is
`<?php if (!is_null($this->item->publish_down)`
Don't know what is better.

Comment (SharkyKZ) [reply layouts/joomla/content/blog_style_default_item_title.php:45]: Strict `null` comparison is generally faster than `is_null()`. There's PHP-CS-Fixer rule for removing `is_null()` uses which we could apply eventually.

**Comment by infograf768** [comment]
I have tested this item :white_check_mark: successfully on 34abc21ec57308cf77fcfb99f70bb0e0a0df0c16<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/26516">issues.joomla.org/tracker/joomla-cms/26516</a>.</sub>

File: layouts/joomla/content/blog_style_default_item_title.php (hunk 1 above)
Comment (richard67) [reply layouts/joomla/content/blog_style_default_item_title.php:45]: I didn't know that. Thanks for the info.

**Comment by alikon** [comment]
I have tested this item :white_check_mark: successfully on 34abc21ec57308cf77fcfb99f70bb0e0a0df0c16<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/26516">issues.joomla.org/tracker/joomla-cms/26516</a>.</sub>

**Comment by alikon** [comment]
RTC<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/26516">issues.joomla.org/tracker/joomla-cms/26516</a>.</sub>

**Comment by wilsonge** [comment]
Thanks!

**Pull request merged**